hermeto/core/package_managers/python/pip/markers.py

```
@@ -0,0 +1,299 @@
+            # touches pinned, concrete values.
+            if marker.evaluate(env, context="requirement"):
+                return True
+        except Exception:
+            # Defensive: the gate should make evaluation total, so any error here
+            # is unexpected -- fail open rather than risk a wrong skip.
+            log.debug(
+                "Keeping %r: could not evaluate its environment marker (fail-open)",
+                requirement.download_line,
+                exc_info=True,
+            )
+            return True
```

Ruff (0.16.1)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)